src/config.ts

```diff
@@ -1,42 +1,35 @@
-import {
-  CSSDeclaration,
-  OneDArray,
-  ProtoExtends,
-  TColumn,
-  TData,
-} from './types';
+import { CSSDeclaration, OneDArray, Status, TColumn, TData } from './types';
 import Storage from './storage/storage';
 import Pipeline from './pipeline/pipeline';
 import Tabular from './tabular';
 import { Search, SearchConfig } from './view/plugin/search/search';
 import { Pagination, PaginationConfig } from './view/plugin/pagination';
 import Header from './header';
 import { ServerStorageOptions } from './storage/server';
-import Dispatcher from './util/dispatcher';
 import { GenericSortConfig } from './view/plugin/sort/sort';
 import { Language, Translator } from './i18n/language';
-import { Component, ComponentChild, createRef, RefObject } from 'preact';
+import { ComponentChild, createContext, createRef, RefObject } from 'preact';
 import StorageUtils from './storage/storageUtils';
 import PipelineUtils from './pipeline/pipelineUtils';
 import { EventEmitter } from './util/eventEmitter';
 import { GridEvents } from './events';
 import { PluginManager, PluginPosition, Plugin } from './plugin';
 import Grid from './grid';
+import { Store } from './state/store';
+
+export const ConfigContext = createContext(null);
 
-// Config type used internally
 export interface Config {
   // a reference to the current Grid.js instance
   instance: Grid;
+  store: Store<any>;
```

Comment:
Can we use the Return type here to solve that issue ? https://www.typescriptlang.org/docs/handbook/utility-types.html#returntypetype

I haven't looked too hard into it, but if the store is created with a function that might be the easy way.
Or if you have an initial store, `typeof myInitialStore` would do the trick too.

Reply:
EDIT: Disregard the previous comment I didn't realise it was a generic.

I think you simply can set it to `unknown`? 
Or if it has constraints, such as it has to be an object, the `extends` would help but as far I can see it has no constraints.

Reply:
Good point!